pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -57,9 +57,9 @@ const MaxReadThenCreateRetries = 30
 
 // Abstraction over AWS, to allow mocking/other implementations
 type AWSServices interface {
-	Compute(region string) (EC2, error)
-	LoadBalancing(region string) (ELB, error)
-	Autoscaling(region string) (ASG, error)
+	Compute(region string, endpoint string) (EC2, error)
+	LoadBalancing(region string, endpoint string) (ELB, error)
+	Autoscaling(region string, endpoint string) (ASG, error)
```

Comment:
I'm almost wondering if we should just pass in *aws.Config.  For now it's fine, but I think if we get to 4 arguments we should switch!